src/Framework/Sdk/SdkResultItem.cs

```diff
@@ -55,14 +60,14 @@ public override bool Equals(object obj)
         public override int GetHashCode()
         {
             int hashCode = -849885975;
-            hashCode = (hashCode * -1521134295) + EqualityComparer<string>.Default.GetHashCode(ItemSpec);
+            hashCode = hashCode ^ ItemSpec.GetHashCode();
 
-            if (Metadata != null)
+            if (Metadata != null && Metadata.Count > 0)
             {
                 foreach (var kvp in Metadata)
                 {
-                    hashCode = (hashCode * -1521134295) + kvp.Key.GetHashCode();
-                    hashCode = (hashCode * -1521134295) + kvp.Value.GetHashCode();
+                    hashCode = hashCode ^ kvp.Key.GetHashCode();
+                    hashCode = hashCode ^ kvp.Value.GetHashCode();
```

Comment:
Likewise here make sure the key/value relationship is order-preserving, while making the inclusion in the overall hashcode order-agnostic. That way these two items compare different:

```
ItemSpec = "ItemSpec"
Metadata = {
  ["Key"] = "value"
}
```

and

```
ItemSpec = "ItemSpec"
Metadata = {
  ["value"] = "Key"
}
```

Reply:
Just to confirm, the order of the Key in the keys are still ignoring the order, right?
Ok, I'll make the GetHashTable order-preserving for Key-Value more granular.
I'm going to update that including the unit test additions.

Reply:
Correct; the order of the key-value pairs in the enumeration should not matter, but within each one the key vs value should.

Reply:
Done fixing the Key-Value of GetHashCode. Currently adding "K" in hash code gen for Key, and adding "V" for hashcode gen for Value. Therefore Key and Value will always be different when we are getting hashcode of key and value of the enumerated Key-Value pair.

https://github.com/eriawan/msbuild/blob/fix-SdkResultItem-equality/src/Framework/Sdk/SdkResultItem.cs#L66-L70

Reply:
@rainersigwald 

Done applying the suggestion and also in sync with feedback from others.